### PR TITLE
Linebreaking

### PR DESCRIPTION
**Pull request by khalidalqassabi**
based on https://github.com/HOST-Oman/libraqm/pull/58 plus fixing error in the right alignment the text get shifted more than it's supposed to. However, I faced a problem with the tests. All of them are not function at all.(fail) and it reach to multi-fonts then it crash.

@khaledhosny  can you help me

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -3,8 +3,8 @@ AC_INIT([Raqm],
         [https://github.com/HOST-Oman/libraqm/issues],
         [raqm],
         [https://github.com/HOST-Oman/libraqm/])
-
-AM_INIT_AUTOMAKE([foreign color-tests -Wall -Wno-portability])
+AM_INIT_AUTOMAKE([subdir-objects])
+#AM_INIT_AUTOMAKE([foreign color-tests -Wall -Wno-portability])
```
Comment (khaledhosny) [inline configure.ac]: Just add `subdir-objects` to the existing automake options, don’t replace it.

File: src/raqm.c [hunk 2]
```diff
@@ -31,6 +31,8 @@
 #include <hb.h>
 #include <hb-ft.h>
 
+#include <ucdn/ucdn.h>
```
Comment (khaledhosny) [inline src/raqm.c]: It should be `include "ucdn/ucdn.h"` since it is not a system header.

**Comment by khaledhosny** [comment]
Please do a clean rebase without `Merge branch …` commits.

**Comment by khaledhosny** [comment]
The ucdn files should be added as a git submodule, instead of copying them into the repository.

File: src/raqm.c [hunk 3]
```diff
@@ -834,6 +834,7 @@ enum break_action {
         RAQM_COMBINING_INDIRECT_BREAK,		// # in table, 	cc in array
         RAQM_COMBINING_PROHIBITED_BREAK,  	// @ in table 	CC in array
         RAQM_PROHIBITED_BREAK,             	// ^ in table, 	XX in array
+        RAQM_UNDEFINE_BREAK,             	// ^ in table, 	XX in array
```
Comment (khaledhosny) [inline src/raqm.c]: This comment is wrong.

**Comment by khaledhosny** [comment]
Please rebase on master to fix Travis build.

**Comment by khalidalqassabi** [comment]
updated, but it still not working every change required is done  
can you help me 
@khaledhosny

**Comment by khaledhosny** [comment]
You should not have merged all the commits together, we now lost track who wrote what. You should have just done `git rebase master`.

**Comment by khaledhosny** [comment]
Please configure with `--enable-gtk-doc` and fix gtkdoc-check.test failures.

File: src/raqm.h [hunk 4]
```diff
@@ -81,10 +98,14 @@ typedef enum
  */
 typedef struct raqm_glyph_t {
     unsigned int index;
+    int visual_index;
+    int line;
```
Comment (khaledhosny) [inline src/raqm.h]: Move those two members to the end of the struct and mark them private for GTK-Doc.

**Comment by khaledhosny** [comment]
See #65.

**Pull request closed without merging**